src/main/java/ca/on/oicr/pineryreports/reports/impl/LanesBillingReport.java

```diff
@@ -0,0 +1,464 @@
+    // samples and instruments and instrument models will come in handy
+    Map<String, SampleDto> samplesById = mapSamplesById(pinery.getSample().all());
+    Map<Integer, InstrumentDto> instrumentsById = pinery.getInstrument().all().stream()
+        .collect(Collectors.toMap(InstrumentDto::getId, dto->dto));
+    Map<Integer, InstrumentModelDto> instrumentModelsById = pinery.getInstrumentModel().all().stream()
+        .collect(Collectors.toMap(InstrumentModelDto::getId, dto->dto));
+    // filter runs within the date range
+    Set<RunDto> newRuns = pinery.getSequencerRun().all().stream().filter(byEndedBetween(start, end)).collect(Collectors.toSet());
+    Set<RunDto> failedRuns = newRuns.stream().filter(run -> "Failed".equals(run.getState())).collect(Collectors.toSet());
+    newRuns.removeAll(failedRuns);
+    
+    completed = getReportAndSummaryData(newRuns, samplesById, instrumentsById, instrumentModelsById);
+    failed = getReportAndSummaryData(failedRuns, samplesById, instrumentsById, instrumentModelsById);
+  }
+  
+  private ReportObject getReportAndSummaryData(Collection<RunDto> runs, Map<String, SampleDto> samplesById,
+      Map<Integer, InstrumentDto> instrumentsById, Map<Integer, InstrumentModelDto> instrumentModelsById) {
+    List<DetailedObject> detailedData = new ArrayList<>();
+    Map<String, Map<String, BigDecimal>> summaryData = new TreeMap<>();
+    for (RunDto run : runs) {
+      String runName = run.getName();
+      String runStatus = run.getState();
+      String instrumentName = getInstrumentName(run.getInstrumentId(), instrumentsById);
+      String completionDate = run.getCompletionDate();
+      String instrumentModel = getInstrumentModel(run.getInstrumentId(), instrumentsById, instrumentModelsById);
+      if (run.getPositions() == null) continue;; 
+      for (RunDtoPosition lane : run.getPositions()) {
+        int laneNumber = lane.getPosition();
+        int samplesInLane = lane.getSamples() == null ? 0 : lane.getSamples().size();
+        boolean dnaInLane = false;
+        boolean rnaInLane = false;
+        // projectName : numLibsFromProjInLane
+        Map<String, Integer> projectsInLane = new HashMap<>();
+        if (lane.getSamples() == null) continue;
+        for (RunDtoSample sam : lane.getSamples()) {
+          SampleDto dilution = samplesById.get(sam.getId());
+          if (isRnaLibrary(dilution, samplesById)) {
+            rnaInLane = true;
+          } else {
+            dnaInLane = true;
+          }
+          if (projectsInLane.containsKey(dilution.getProjectName())) {
+            // increment project libraries
+            projectsInLane.merge(dilution.getProjectName(), 1, Integer::sum);
+          } else {
+            // add project with library to map
+            projectsInLane.put(dilution.getProjectName(), 1);
+          }
+        }
+        String laneContents = dnaInLane 
+            ? (rnaInLane ? MIXED_LANE : DNA_LANE) 
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)